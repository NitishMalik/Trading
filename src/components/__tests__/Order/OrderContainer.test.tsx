import React from 'react';
import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import OrderConfirmation from '../../Order/OrderContainer';

describe('tests for Login component', () => {
  it('generates snapshot for Order component', () => {
    const {container} = render(<OrderConfirmation />);
    expect(container).toMatchSnapshot();
  });

  it('should not able to enter stock name', () => {
    // Arrange
    const stockName = 'RIL';

    // Act
    const {getByTestId} = render(<OrderConfirmation />);
    const stockNameEle = getByTestId('stock-name-ele');
    userEvent.type(stockNameEle, stockName);

    // Assert
    expect((getByTestId('stock-name-ele') as HTMLInputElement).value).not.toBe(
      stockName,
    );
  });

  it('should able to enter stock price', () => {
    // Arrange
    const stockPrice = '595';

    // Act
    const {getByTestId} = render(<OrderConfirmation />);
    const stockPriceEle = getByTestId('stock-price-ele');
    userEvent.type(stockPriceEle, stockPrice);

    // Assert
    expect((getByTestId('stock-price-ele') as HTMLInputElement).value).toBe(
      stockPrice,
    );
  });

  it('should able to enter stock quantity', () => {
    // Arrange
    const stockQuantity = '250';

    // Act
    const {getByTestId} = render(<OrderConfirmation />);
    const stockQuantityEle = getByTestId('quantity-ele');
    userEvent.type(stockQuantityEle, stockQuantity);

    // Assert
    expect((getByTestId('quantity-ele') as HTMLInputElement).value).toBe(
      stockQuantity,
    );
  });

  it('should able to select transaction type', () => {
    // Act
    const {getByTestId} = render(<OrderConfirmation />);
    const transactionTypeEle = getByTestId('transaction-buy');
    userEvent.click(transactionTypeEle);

    // Assert
    expect((getByTestId('transaction-buy') as HTMLInputElement).value).toBe(
      'on',
    );
  });

  it('should able to click on place order', () => {
    // Act
    const {getByTestId} = render(<OrderConfirmation />);
    const stockNameEle = getByTestId('place-order-button');
    userEvent.click(stockNameEle);

    // Assert
    expect(
      getByTestId('transaction-buy') as HTMLInputElement,
    ).not.toBeDisabled();
  });
});
