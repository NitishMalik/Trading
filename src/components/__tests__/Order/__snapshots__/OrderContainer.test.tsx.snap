// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Login component generates snapshot for Order component 1`] = `
<div>
  <div
    class="order-screen container"
  >
    <form
      class=""
    >
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
          for="stockName"
        >
          Stock Name
        </label>
        <div
          class="col-sm-10"
        >
          <input
            class="bold form-control-plaintext"
            data-testid="stock-name-ele"
            id="stockName"
            readonly=""
            value="Adani Green"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
          for="stockPrice"
        >
          Stock Price
        </label>
        <div
          class="col-sm-5"
        >
          <input
            class="form-control"
            data-testid="stock-price-ele"
            id="stockPrice"
            placeholder="0.0"
            type="text"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
          for="formPlaintextPassword"
        >
          Quantity
        </label>
        <div
          class="col-sm-5"
        >
          <input
            class="form-control"
            data-testid="quantity-ele"
            id="formPlaintextPassword"
            placeholder="0"
            type="text"
          />
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
          for="formPlaintextPassword"
        >
          Transaction Type
        </label>
        <div
          class="no-padding--left col-sm-2"
        >
          <div
            class="form-check form-check-inline"
          >
            <input
              class="form-check-input"
              data-testid="transaction-buy"
              id="transaction-type--buy"
              type="radio"
            />
            <label
              class="form-check-label"
              for="transaction-type--buy"
              title=""
            >
              Buy
            </label>
          </div>
          <div
            class="form-check form-check-inline"
          >
            <input
              class="form-check-input"
              data-testid="transaction-sell"
              id="transaction-type--sell"
              type="radio"
            />
            <label
              class="form-check-label"
              for="transaction-type--sell"
              title=""
            >
              Sell
            </label>
          </div>
        </div>
      </div>
      <div
        class="form-group row"
      >
        <label
          class="form-label col-form-label col-sm-2"
        />
        <div
          class="no-padding--left col-sm-2"
        >
          <button
            class="btn btn-primary"
            data-testid="place-order-button"
            type="submit"
          >
            Place Order
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
